### ISSUE TITLE
About ng_on_watch error

### ISSUE THREAD

**Issue by albertosantini** [Gillespie59/eslint-plugin-angular#114]
I am confused.

It (the watcher) will effectively get removed once the scope is destroyed.
No need to destroy it manually. At least, in the main cases.

The details about ng_on_watch refer to Y035, but that link is about "Defer Controller Logic to Services".

Am I missing anything?

**Comment by Gillespie59** [comment]
It looks like the Y035 link in at the wrong place. 

You should manually the method to unbind your $watcher during the $destroy event

```
var watcher = $scope.$watch(...);
$scope.$on('$destroy', function(){
  watch();
});
```

**Comment by albertosantini** [comment]
I don't see any recommendation about destroying manually the watcher.
I gave a look in John Papa's and Todd Motto's guides.

Maybe we may change the default.

**Comment by Gillespie59** [comment]
I do not remember where I found this best practice. 

But you should listen to the $destroy event in order to unbind $watch watcher and stop timers for example.

Any remarks from somebody else ? ;)

**Comment by davidmason** [comment]
I have just been looking into this, still more looking to do. Everything I read says that `$scope.$on` and `$scope.$watch` will be automatically unbound when the controller is destroyed.

If a controller is registering a listener on `$rootScope` it should probably be manually destroyed in `"$destroy"` since root scope will outlive all the controllers.

I cannot see anything on https://github.com/johnpapa/angular-styleguide about this rule specifically. The closest I can find is a note in [Y070](https://github.com/johnpapa/angular-styleguide#style-y070):

> Note: "**Best Practice**: Directives should clean up after themselves. You can use `element.on('$destroy', ...)` or `scope.$on('$destroy', ...)` to run a clean-up function when the directive is removed" ... from the Angular documentation.

I will look into this more and see what changes seem sensible.

**Comment by taybin** [comment]
http://stackoverflow.com/questions/26983696/angularjs-does-destroy-remove-event-listeners

It looks like it's only necessary for cleaning up $on and $watch on $rootScope, not $scope.

**Comment by Gillespie59** [comment]
I have just published a version 0.1.2 with a fix for this issue. Thanks

**Comment by albertosantini** [comment]
Tested. Great!

Thanks for your effort.
